plugins/browser-plugin-media-tracking/src/player.ts

```diff
@@ -0,0 +1,174 @@
+import { v4 as uuidv4 } from 'uuid';
+import {
+  startMediaTracking,
+  MediaEventType,
+  trackMediaPlay,
+  trackMediaPause,
+  trackMediaEnd,
+  trackMediaSeekEnd,
+  trackMediaPlaybackRateChange,
+  trackMediaVolumeChange,
+  trackMediaFullscreenChange,
+  trackMediaPictureInPictureChange,
+  trackMediaBufferStart,
+  trackMediaBufferEnd,
+  trackMediaError,
+  updateMediaTracking,
+  trackMediaReady,
+} from '@snowplow/browser-plugin-media';
+import { MediaPlayerUpdate, MediaType } from '@snowplow/browser-plugin-media/src/types';
+import { ElementConfig } from './config';
+import { getDuration, isFullScreen, parseVolume } from './helperFunctions';
+import { buildHTMLMediaElementEntity, buildHTMLVideoElementEntity } from './entities';
+import { LOG, SelfDescribingJson } from '@snowplow/tracker-core';
+
+type CommonMediaPlayerUpdate = Omit<MediaPlayerUpdate, 'fullscreen' | 'pictureInPicture' | 'quality'>;
+type VideoMediaPlayerUpdate = Pick<MediaPlayerUpdate, 'fullscreen' | 'pictureInPicture'>;
+
+function updatePlayer(el: HTMLMediaElement): MediaPlayerUpdate {
+  const common: CommonMediaPlayerUpdate = {
+    currentTime: el.currentTime || 0,
+    duration: getDuration(el),
+    ended: el.ended,
+    livestream: el.duration === Infinity,
+    loop: el.loop,
+    mediaType: el.tagName === 'VIDEO' ? MediaType.Video : MediaType.Audio,
+    muted: el.muted,
+    paused: el.paused,
+    volume: parseVolume(el.volume),
+    playbackRate: el.playbackRate,
+  };
+
+  const video: VideoMediaPlayerUpdate =
+    el instanceof HTMLVideoElement
+      ? {
+          fullscreen: isFullScreen(el),
+          pictureInPicture: document.pictureInPictureElement === el,
+        }
+      : {};
+
+  return { ...common, ...video };
+}
+
+function htmlContext(el: HTMLMediaElement): (() => SelfDescribingJson)[] {
+  const context = [() => buildHTMLMediaElementEntity(el)];
+
+  if (el instanceof HTMLVideoElement) {
+    context.push(() => buildHTMLVideoElementEntity(el));
+  }
+
+  return context;
+}
+
+export function setUpListeners(config: ElementConfig) {
+  const { video } = config;
+  const id = uuidv4();
+
+  startMediaTracking({
+    ...config,
+    id,
+    player: updatePlayer(video),
+    context: (config.context ?? []).concat(htmlContext(video)),
+  });
+
+  // If metadata is already loaded, we can track the ready event immediately
+  if (video.readyState > 0) {
+    trackMediaReady({ id, player: updatePlayer(video) });
+  } else {
+    video.addEventListener('loadedmetadata', () => {
+      trackMediaReady({ id, player: updatePlayer(video) });
+    });
+  }
+
+  video.addEventListener('timeupdate', () => {
+    updateMediaTracking({ id, player: updatePlayer(video) });
+  });
+
+  let isWaiting = false;
+
+  for (const event of config.captureEvents ?? []) {
```

Comment:
So by default we don't capture any events? Shouldn't we have some default list of events to capture?

Reply:
It might be better to subscribe to all the events regardless of the `captureEvents` setting and let the underlying media package filter them using the `captureEvents` property. This will let the package update the media session entity correctly (based on the play, pause, volume events), but it won't track events that the user doesn't want (by default it will track all events).

Reply:
Sounds good, `captureEvents` is already passed into the `startMediaTracking` config, so I've just removed the switch so all events listeners are added.